### ISSUE TITLE
Inet data type doesn't appear to be supported

### ISSUE THREAD

**Issue by jakeatwork** [plataformatec/simple_form#1241]
In working through an issue I found with the sabisu gem, it appears that simple_form doesn't natively support the inet datatype new in rails 4 for IP addresses. They used to be stored as strings/varchar(255).

Here's the issue I was having with sabisu gem and my stackoverflow answer for the issue:

https://github.com/IcaliaLabs/sabisu-rails/issues/20

http://stackoverflow.com/questions/29340177/cant-get-sabisu-gem-to-render-shows-no-input-found-for-inet-error/29340178#29340178

And the article that led me to the solution: https://blog.engineyard.com/2013/new-in-rails-4

Not sure if there was some other way to approach, so I just created a new migration to push those fields back to string. So personally I don't need this fixed, but wanted to make sure I shared it.

**Comment by carlosantoniodasilva** [comment]
We could simply map them to strings, as I did in [this experimental branch](https://github.com/plataformatec/simple_form/compare/ca-pg-types) some time ago. Meanwhile you can map those yourself, or pass `as: :string` on these input fields.

**Comment by metaskills** [comment]
Had an issue on the SQL Server adapter since we subclass every ActiveRecord core type with explicit SQL Server types. Based on our chart here [Native Data Type Support](https://github.com/rails-sqlserver/activerecord-sqlserver-adapter#native-data-type-support) perhaps this project can add adapter specific mappings? Here is what I see for ours.

```ruby
# SQL Server Adapter
map_type :ntext, :varchar_max, :nvarchar_max,    to: SimpleForm::Inputs::TextInput
map_type :char, :nchar, :varchar,                to: SimpleForm::Inputs::StringInput
map_type :bigint, :real, :smallmoney, :money,    to: SimpleForm::Inputs::NumericInput
```

**Comment by kwerle** [comment]
My expectation for this problem was that I would simply do
config.input_mappings = { /varchar/ => :string }
and that would take care of the issue (because all I needed to map was varchar).  But the input_mappings code doesn't behave the way I expect - it appends 'input' to the pattern and tries to match that to a class.
Certainly some kind of simple mapping as described above would solve the problem.  I ended up adding a VarcharInput class to SimpleForm (that mostly just subclassed StringInput).  Not hard, but certainly more work than I expected.
